core/src/main/scala/org/apache/spark/eventhubs/common/EventHubsConf.scala

```diff
@@ -0,0 +1,506 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.eventhubs.common
+
+import java.time.Duration
+import java.util.concurrent.ConcurrentHashMap
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.catalyst.util.CaseInsensitiveMap
+
+import scala.collection.JavaConverters._
+import language.implicitConversions
+
+// TODO: Can we catch malformed configs at compile-time?
```

Comment:
Two changes coming to EventHubsConf:
1) An EventHubsConf will have to be constructed with a connection string. The setters for namespace, name, key name, and key will be removed. We expect most people to copy/paste their connection string from the service portal. For those that prefer to do it piecemeal (like I personally do), I'll expose the connection string builder from the new java client which users can use to make their connection string. This makes the configuration process simpler which makes the rest of the code cleaner and less bug-prone. 

2) The new EventHubs Java client will have a new EventPosition abstraction. I'll be removing the setters (and all internals) for seq number, offset, and enqueued time. I'll be replacing that with `setStartingPosition(EventPosition)` and `setStartingPosition(Map[PartitionId -> EventPosition])`. This also will clean things up quite a bit.